homeassistant/components/assist_pipeline/pipeline.py

```diff
@@ -332,12 +332,12 @@ class PipelineRun:
     event_callback: PipelineEventCallback
     language: str = None  # type: ignore[assignment]
     runner_data: Any | None = None
-    stt_provider: stt.SpeechToTextEntity | stt.Provider | None = None
     intent_agent: str | None = None
-    tts_engine: str | None = None
     tts_audio_output: str | None = None
 
     id: str = field(default_factory=ulid_util.ulid)
+    stt_provider: stt.SpeechToTextEntity | stt.Provider = field(init=False)
```

Comment:
Why do we not need to initialize these attributes?

Reply:
OK, I guess that's a bit silly. Changed back to initialize to `None`

Reply:
I think it's ok to wait with initialization if it's a developer error to not initialize them indirectly via some other method call.